### PR TITLE
Edit periods

### PR DESCRIPTION
**Pull request by UndeclaredVariable**
Adds a form for editing periods, and adds a `delete` action to the table.

### CODE REVIEW

File: delete.php [hunk 1]
```diff
@@ -25,9 +25,18 @@
...
+    $dbname = 'cpdlogbook_periods';
+    $conditions = ['id' => $id];
+} else {
+    $dbname = 'cpdlogbook_entries';
+    $conditions = ['id' => $id, 'userid' => $USER->id];
+}
 
 // If the entry doesn't exist.
-$record = $DB->get_record('cpdlogbook_entries', ['id' => $id, 'userid' => $USER->id], '*', MUST_EXIST);
+$record = $DB->get_record($dbname, $conditions, '*', MUST_EXIST);
```
Comment (brendanheywood) [inline delete.php:39]: Generally I would avoid abstracting away the table name, I've move this into two code paths purely make it easier to read and easier to search and grep on later.

File: editperiod.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        unset($fromform->id);
+
+        $fromform->cpdlogbookid = $cm->instance;
+
+        $entryid = $DB->insert_record('cpdlogbook_periods', $fromform, true);
+    } else {
+        // Update the record according to the submitted form data.
+        $fromform->modifieddate = time();
+        $DB->update_record('cpdlogbook_periods', $fromform);
+        $entry = $DB->get_record('cpdlogbook_periods', ['id' => $fromform->id]);
```
Comment (brendanheywood) [inline editperiod.php:75]: Is this line needed?

Comment (UndeclaredVariable) [reply editperiod.php:74]: That's a remnant from `edit.php`. It's not needed for periods, so I've removed it now.

File: classes/form/edit_period.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * The definition of the form. Used to add elements and rules to the form.
+     *
+     * @throws \coding_exception
+     */
+    public function definition() {
+        $mform = $this->_form;
+
+        $mform->addElement('date_selector', 'startdate', get_string('startdate', 'mod_cpdlogbook'));
+        $mform->setType('startdate', PARAM_TEXT);
```
Comment (brendanheywood) [inline classes/form/edit_period.php:47]: this is probably PARAM_TEXT or possibly can just be omitted

Comment (brendanheywood) [reply classes/form/edit_period.php:47]: apart from this the rest is good to go, so feel free to merge away when its green again

**Pull request merged**